policy/lib/purl.rego

```diff
@@ -0,0 +1,26 @@
+package lib.purl
+
+import rego.v1
+
+# Extract the repoid
+repoid(purl) := _named_qualifier("repository_id", purl)
+
+#
+# Extract a named quantifier from a purl.
+# (Quantifiers are similar to url params.)
+# Let's fall back to returning null if the quantifier is not present.
+#
+_named_qualifier(key, purl) := result if {
+	# Sanity check I guess
+	ec.purl.is_valid(purl)
```

Comment:
Not sure if this is actually needed. Won't `parse` fail anyways if it is not a valid purl?

Reply:
The difference is whether or not ec-cli produces an error with log.Errorf, (if https://github.com/enterprise-contract/ec-cli/pull/1923 is merged.) My thinking is I'd rather not produce an error, so that's why I'm doing is_valid first.